### PR TITLE
gitmodules: add ext/boost_1_82_0 & build.boost.sh: macos x84_64 tweak

### PR DESCRIPTION
**Pull request by RandyMcMillan**

### CODE REVIEW

File: build.boost.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+git submodule update --init --recursive ext/boost_1_82_0
```
Comment (RandyMcMillan) [inline build.boost.sh:3]: we add boost_1_82_0 submodule - branch/tag boost-1.82.0 per your established boost version.

NOTE: the brew build takes longer than this scripted build - I actually like building it this way better...

File: build.boost.sh [hunk 2]
```diff
@@ -8,8 +10,8 @@ if [[ "$OSTYPE" == "msys" ]]; then
 ./b2 --prefix=.  --layout=versioned --toolset=msvc-14.3 address-model=64 architecture=x86 variant=debug threading=multi link=static runtime-link=shared install
 elif [[ "$OSTYPE" == "darwin"* ]]; then
 
-./bootstrap.sh
-./b2 --prefix=. variant=release 
+./bootstrap.sh --prefix=`pwd`
```
Comment (RandyMcMillan) [inline build.boost.sh:13]: Using `pwd` works consistently. The . was acting strange...

Comment (RandyMcMillan) [reply build.boost.sh:13]: NOTE: AFAIK - the --prefix=`pwd` invokes the bootstrap.sh to create an include/ folder - which ./b2  --prefix=`pwd` detects consistently in the next line.

**Comment by pedro-vicente** [comment]
With this update, I get on MacOS

```
./build.boost.sh
Submodule 'ext/boost_1_82_0' (git@github.com:boostorg/boost.git) registered for path 'ext/boost_1_82_0'
Cloning into '/Users/pvn/github/nostr_client_relay/ext/boost_1_82_0'...
git@github.com: Permission denied (publickey).
```


No sure if it;s because there is an attempt to access SSH URL (Read/Write) instead of Git Read-Only URL , as explained here ?

https://stackoverflow.com/questions/2643502/git-how-to-solve-permission-denied-publickey-error-when-using-git

**Comment by RandyMcMillan** [comment]
yeah you or i can change it to the https:// url

**Comment by RandyMcMillan** [comment]
noted - i am used to using ssh for every thing...

**Pull request merged**